src/lib/customclasspath/RegistryJsonHandler.ts

```diff
@@ -0,0 +1,118 @@
+import fs = require('fs');
+import path = require('path');
+import util = require('util');
+
+
+export const CUSTOM_CLASSPATH_REGISTER = path.join('pmd-cataloger', 'catalogs', '.CustomPaths.json');
+export class RegistryJsonHandler {
+
+    private jsonFileOperator: JsonFileOperator;
+
+    constructor() {
+        this.jsonFileOperator = new JsonFileOperator();
+    }
+
+    async readCurrentEntries(): Promise<Map<string, Map<string, string[]>>> {
+        const jsonData = await this.jsonFileOperator.readJsonFile();
+        const mapFromFile = this.jsonToMap(jsonData);
+
+        return mapFromFile;
+    }
+
+    async updateEntries(engineToLanguageMap: Map<string, Map<string, string[]>>) {
```

Comment:
As we discussed on the call, you could make the top-level object a JSON whose keys are the Engine enums and whose values are Maps, and you could define an alias for that object so the code is easier to read.

Reply:
This is a good idea. I have a few more thoughts on regrouping the related methods as well. Will pick it up as another iteration of changes on this code.